[android] Add SDK32

tsapeta opened the pull request:
# Why

Step of https://github.com/expo/expo/issues/2969

# How

Followed https://github.com/expo/universe/blob/master/internal-docs/monthly-release.md#android guide.

# Test Plan

- [x] Expo Client runs
- [x] Sandbox with `sdkVersion = 32.0.0` runs

tsapeta committed:
[sdk32] Versioned SDK32 on Android

tsapeta committed:
[sdk] fix StoreReview throwing errors on Android

sjchmiela commented on android/expoview/src/main/java/host/exp/exponent/experience/MultipleVersionReactNativeActivity.java:
```diff
@@ -3,7 +3,7 @@
 
 // Implement for each version.
-public class MultipleVersionReactNativeActivity extends ReactNativeActivity implements
+public class MultipleVersionReactNativeActivity extends ReactNativeActivity implements abi32_0_0.com.facebook.react.modules.core.DefaultHardwareBackBtnHandler,
```
Move this line down in between `BEGIN_SDK_32` and `END` (or sth like that)

sjchmiela reviewed (changes requested):
Thanks! Just one comment

tsapeta committed:
[android] fix MultipleVersionReactNativeActivity

Commit:
Fixed type of calendarId

tsapeta committed:
[android] update exponent view for SDK32

tsapeta committed:
[android] restore comments in build.gradle

tsapeta committed:
[android] add stripe-android to app/build.gradle

tsapeta committed:
[android] fix comments in expoview/build.gradle

tsapeta committed:
[android] fix infer-annotation being imported from incorrect package

tsapeta committed:
[android] add missing comments in expoview/build.gradle

tsapeta committed:
[android] Move NotificationActionCenter and DB models to kernel (#3012)

# Why

Because they cannot be versioned as the database models are being generated by our dependency `dbflow` with the same class names and the same package names.
As a result, when I've versioned the code for SDK32, I've got such error on CI:

```
Warning: Exception while processing task java.io.IOException: Can't write [/root/expo/android/app/build/intermediates/transforms/proguard/prodKernel/release/0.jar] (Can't read [/root/expo/android/expoview/build/intermediates/intermediate-jars/release/classes.jar(;;;;;;;**.class)] (Duplicate jar entry [com/raizlabs/android/dbflow/config/c.class]))
Thread(Tasks limiter_1): destruction

> Task :app:transformClassesAndResourcesWithProguardForProdKernelRelease FAILED

FAILURE: Build failed with an exception.
```

# How

Moved `NotificationActionCenter`, `ActionDatabase`, `ActionTable` and `IntentProvider` to the package that is not being versioned.

# Test Plan

Expo Client with versioned SDK32 builds locally.
